### PR TITLE
Fixes Quick Draw 

### LINKED ISSUE #3662: Custap Berry does not activate when the Player switches out
### Description

If the foe's Pokemon is holding a Custap Berry and is in range of its activation, it does not activate when the Player switches out and can still be used in future turns. 

The expected behaviour is that the Custap Berry activates and gets consumed when the Player switches out.

### Version

upcoming (Edge)

### Upcoming Version

Custap Berry wouldn't trigger if the opposing Pokémon switched out.

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Quick Draw was not functioning correctly. Even though the ability triggered it didn't attack first sometimes. Additionally the user randomly attacked first without the ability triggering. For some reason tests don't catch it so it has to be tested manually. I've ran it manually a lot of times but it would be good to have confirmation that it is fixed now.

Recording with Slowbro attacking first without the ability triggering.

https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/ea319ffc-690f-426a-bd46-65686709b48a

Also I added a test for Quick Claw and removed `EWRAM_DATA u16 gRandomTurnNumber`. The only place it was used was the Quick Claw check which I replaced with `RandomPercentage` 

Fixes #3662

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Drafting this until I figure out what is going on

**Comment by AlexOn1ine** [comment]
So I accidentally fixed it. When I was fixing issue #3662 I realized my fix might apply here as well and it seems like it did.

I moved all the Custap, Quick Draw, Quick Claw checks out of the function to a new one so it's only called from `SetActionsAndBattlersTurnOrder`

File: src/battle_main.c [hunk 1]
```diff
@@ -5115,7 +5092,37 @@ static void TryChangeTurnOrder(void)
...
+    u32 holdEffectBattler2 = GetBattlerHoldEffect(battler2, TRUE);
+    u32 ability2 = GetBattlerAbility(battler2);
+
+    // Battler 1
+    // Quick Draw
+    if (ability1 == ABILITY_QUICK_DRAW && !IS_MOVE_STATUS(gChosenMoveByBattler[battler1]) && RandomPercentage(RNG_QUICK_DRAW, 30))
+        gProtectStructs[battler1].quickDraw = TRUE;
+    // Quick Claw and Custap Berry
+    if (!gProtectStructs[battler1].quickDraw
+     && ((holdEffectBattler1 == HOLD_EFFECT_QUICK_CLAW && RandomPercentage(RNG_QUICK_CLAW,10))
```
Comment (Bassoonian) [inline src/battle_main.c:5110]:
```suggestion
     && ((holdEffectBattler1 == HOLD_EFFECT_QUICK_CLAW && RandomPercentage(RNG_QUICK_CLAW, 10))
```

File: src/battle_main.c [hunk 2]
```diff
@@ -5115,7 +5092,37 @@ static void TryChangeTurnOrder(void)
...
+     || (holdEffectBattler1 == HOLD_EFFECT_CUSTAP_BERRY && HasEnoughHpToEatBerry(battler1, 4, gBattleMons[battler1].item))))
+        gProtectStructs[battler1].usedCustapBerry = TRUE;
+
+    // Battler 2
+    // Quick Draw
+    if (ability2 == ABILITY_QUICK_DRAW && !IS_MOVE_STATUS(gChosenMoveByBattler[battler2]) && RandomPercentage(RNG_QUICK_DRAW, 30))
+        gProtectStructs[battler2].quickDraw = TRUE;
+    // Quick Claw and Custap Berry
+    if (!gProtectStructs[battler2].quickDraw
+     && ((holdEffectBattler2 == HOLD_EFFECT_QUICK_CLAW && RandomPercentage(RNG_QUICK_CLAW,10))
```
Comment (Bassoonian) [inline src/battle_main.c:5120]:
```suggestion
     && ((holdEffectBattler2 == HOLD_EFFECT_QUICK_CLAW && RandomPercentage(RNG_QUICK_CLAW, 10))
```

**Review by Bassoonian — CHANGES_REQUESTED** [review]

File: src/battle_main.c (hunk 1 above)
Comment (AlexOn1ine) [reply src/battle_main.c]: fixed.

**Review by Bassoonian — APPROVED** [review]

File: src/battle_main.c [hunk 3]
```diff
@@ -5115,7 +5092,37 @@ static void TryChangeTurnOrder(void)
...
+    u32 holdEffectBattler2 = GetBattlerHoldEffect(battler2, TRUE);
+    u32 ability2 = GetBattlerAbility(battler2);
+
+    // Battler 1
+    // Quick Draw
+    if (ability1 == ABILITY_QUICK_DRAW && !IS_MOVE_STATUS(gChosenMoveByBattler[battler1]) && RandomPercentage(RNG_QUICK_DRAW, 30))
+        gProtectStructs[battler1].quickDraw = TRUE;
+    // Quick Claw and Custap Berry
+    if (!gProtectStructs[battler1].quickDraw
+     && ((holdEffectBattler1 == HOLD_EFFECT_QUICK_CLAW && RandomPercentage(RNG_QUICK_CLAW, 10))
```
Comment (ghoulslash) [inline src/battle_main.c:5110]: Quick Claw has a 20% chance of boosting priority

Comment (AlexOn1ine) [reply src/battle_main.c:5110]: Oops, #3737

**Pull request merged**